From f74f123b50a93372399cf7e863042d5dc4d21d2c Mon Sep 17 00:00:00 2001
From: Peter Chiu <chui-hao.chiu@mediatek.com>
Date: Thu, 10 Jul 2025 13:34:49 +0800
Subject: [PATCH 007/119] mtk: mt76: mt7996: init max_ap_assoc_sta to set
 maximum stations

Init max_ap_assoc_sta according to HW capability.
Verified by 'hostapd_cli status driver'

Signed-off-by: Peter Chiu <chui-hao.chiu@mediatek.com>
---
 mt7996/init.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mt7996/init.c b/mt7996/init.c
index f9ee8969..7eade530 100644
--- a/mt7996/init.c
+++ b/mt7996/init.c
@@ -488,6 +488,7 @@ mt7996_init_wiphy(struct ieee80211_hw *hw, struct mtk_wed_device *wed)
 	wiphy->mbssid_max_interfaces = 16;
 	wiphy->iftype_ext_capab = iftypes_ext_capa;
 	wiphy->num_iftype_ext_capab = ARRAY_SIZE(iftypes_ext_capa);
+	wiphy->max_ap_assoc_sta = mt7996_wtbl_size(dev) - MT7996_WTBL_BMC_SIZE - 1;
 
 	wiphy_ext_feature_set(wiphy, NL80211_EXT_FEATURE_BSS_COLOR);
 	wiphy_ext_feature_set(wiphy, NL80211_EXT_FEATURE_VHT_IBSS);
-- 
2.45.2

